File: src/main/java/uk/yermak/audiobookconverter/MediaInfoBean.java
package uk.yermak.audiobookconverter;

import java.util.List;
import java.util.Objects;

public class MediaInfoBean extends MediaInfoOrganiser implements MediaInfo {

    private final String fileName;
    private int channels;
    private int frequency;
    private int bitrate;
    private long duration;
    private AudioBookInfo bookInfo;
    private ArtWork artWork;
    private String codec;

    public void setChannels(final int channels) {
        this.channels = channels;
    }

    public void setFrequency(final int frequency) {
        this.frequency = frequency;
    }

    public void setBitrate(final int bitrate) {
        this.bitrate = bitrate;
    }

    public void setDuration(final long duration) {
        this.duration = duration;
    }

    @Override
    public int getUID() {
        return Objects.hash(fileName);
    }

    @Override
    public String getReference() {
        return getFileName();
    }

    public int getChannels() {
        return this.channels;
    }

    public int getFrequency() {
        return this.frequency;
    }

    public int getBitrate() {
        return this.bitrate;
    }

    @Override
    public String getTitle() {
        return this.getBookInfo().title().get();
    }

    @Override
    public String getDetails() {
        return this.getFileName();
    }

    public long getDuration() {
        return this.duration;
    }

    @Override
    public List<MediaInfo> getMedia() {
        return List.of(this);
    }

    public String getFileName() {
        return this.fileName;
    }

    public void setBookInfo(final AudioBookInfo bookInfo) {
        this.bookInfo = bookInfo;
    }

    public AudioBookInfo getBookInfo() {
        return this.bookInfo;
    }

    public ArtWork getArtWork() {
        return this.artWork;
    }

    public void setArtWork(final ArtWork artWork) {
        this.artWork = artWork;
    }

    public String getCodec() {
        return this.codec;
    }

    public void setCodec(final String codec) {
        this.codec = codec;
    }

    @Override
    public long getOffset() {
        return -1;
    }

    public MediaInfoBean(final String fileName) {
        this.fileName = fileName;
        this.channels = 2;
        this.frequency = 44100;
        this.bitrate = 128000;
        this.duration = 0L;
        this.codec = "";
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof MediaInfo)) return false;
        MediaInfo that = (MediaInfo) o;
        return getReference().equals(that.getReference());
    }

}

File: src/main/scala/uk/yermak/audiobookconverter/MediaInfoBean.scala
package uk.yermak.audiobookconverter

/**
  * Created by Yermak on 02-Jan-18.
  */
class MediaInfoBean(var fileName: String) extends MediaInfo {
  private var channels = 2
  private var frequency = 44100
  private var bitrate = 128000
  private var duration = 0L
  private var bookInfo: AudioBookInfo = null
  private var artWork: ArtWork = null
  private var codec: String = "";

  override def setChannels(channels: Int): Unit = this.channels = channels

  override def setFrequency(frequency: Int): Unit = this.frequency = frequency

  override def setBitrate(bitrate: Int): Unit = this.bitrate = bitrate

  override def setDuration(duration: Long): Unit = this.duration = duration

  override def getChannels: Int = channels

  override def getFrequency: Int = frequency

  override def getBitrate: Int = bitrate

  override def getDuration: Long = duration

  override def getFileName: String = fileName

  override def setBookInfo(bookInfo: AudioBookInfo): Unit = this.bookInfo = bookInfo

  override def getBookInfo: AudioBookInfo = bookInfo

  override def getArtWork: ArtWork = artWork

  override def setArtWork(artWork: ArtWork): Unit = this.artWork = artWork

  override def getCodec: String = codec

  override def setCodec(codec: String): Unit = this.codec = codec
}